### PR TITLE
Fix to custom cigarettes

### LINKED ISSUE #26160: Hand-Rolled Cigarettes Instantly Burn Out
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1641

### Issue Description

Hand-rolled cigarettes instantly burn up and don't transfer any reagents to the smoker.

### What did you expect to happen?

I expected to get a slow transfer of the reagents inside the cigarette to the smoker over the course of 3 to 5 minutes.

### What happened instead?

The cigarette immediately burned up and turned into a cigarette butt/roach on the ground.

### Why is this bad/What are the consequences?

Hand-rolled cigarettes are now completely broken - they cannot be used and smoked. 

### Steps to reproduce the issue.

1. Build a drying rack out of wood.
2. Dry a plant with reagents inside it at the drying rack.
3. Roll up the dried plant in rolling papers. 
4. Equip the resulting cigarette to your mask slot.
5. Light it. 
6. Note that the cigarette has instantly burnt out and left a roach on the ground, with no reagents actually being transferred to you.
7. Optional - dry another copy of the same plant, roll it up, and then grind the rollie in a blender. Note that the reagents inside the cigarette appear just fine when ground, but can't be transferred to you when smoked.

### When did the problem start happening?

This problem didn't exist before the recent PR changing smoking last week. https://github.com/ParadiseSS13/Paradise/pull/25925

### Extra information

I think this issue was introduced in this PR. https://github.com/ParadiseSS13/Paradise/pull/25933 

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by wanderso**
Gave custom-made rollies a smoke_time so that they could be used again as cigarettes.

Changed the smoketime variable to smoke_time, as per snail_case naming convention used elsewhere in the code.

## What Does This PR Do
Fixes #26160. 

Also changed the smoketime variable to be named smoke_time, as per our naming conventions.

## Why It's Good For The Game
Re-enables custom cigarette production.

## Testing
Burnt everything with a smoke_time variable to make sure that it took approximately the same amount of time. Made a bunch of custom cigarettes and found that the smoke_time variable now appropriately causes the cigarette to burn out when it's out of reagents.

## Changelog
:cl:
fix: Fixed hand-rolled cigarettes
/:cl:

### CODE REVIEW

**Comment by wanderso** [comment]
Thanks to DGamerL for making a better 'create custom cigarette' code than I did - I included the code from his refactor in his own pull request after I noticed that he did it at the same time I did.

**Comment by Fordoxia** [comment]
I am reasonably that Midori Tabako (lit. "Green Cigarette") are supposed to contain weed, given both the name and the fact that the description says that they smell funny.

**Review by Fordoxia — APPROVED** [review]
Smokes good to me.

**Review by Sirryan2002 — APPROVED** [review]

**Review by wanderso — COMMENTED** [review]
Removed tab

**Comment by wanderso** [comment]
God. Okay, actually tested this, unlike the other branch this one should actually work. There's probably still bugs in it - some of the stuff seems suspicious - but this actually does allow for the smoking of custom cigarettes

File: code/game/objects/items/weapons/cigs.dm [hunk 1]
```diff
@@ -568,26 +568,23 @@ LIGHTERS ARE IN LIGHTERS.DM
...
-	custom_rollie.desc = "Dried [plant.name] rolled up in a thin piece of paper."
-
-	qdel(plant)
-	qdel(src)
+	if(istype(target, /obj/item/food/snacks/grown))
+		var/obj/item/food/snacks/grown/plant = target
+		if(plant.dry)
+			user.unEquip(target, TRUE)
+			user.unEquip(src, TRUE)
+			var/obj/item/clothing/mask/cigarette/rollie/custom/custom_rollie  = new /obj/item/clothing/mask/cigarette/rollie/custom(user.loc)
```
Comment (Sirryan2002) [inline code/game/objects/items/weapons/cigs.dm:576]:
```suggestion
			var/obj/item/clothing/mask/cigarette/rollie/custom/custom_rollie = new /obj/item/clothing/mask/cigarette/rollie/custom(user.loc)
```

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

**Review by DGamerL — APPROVED** [review]
Please do a merge master!

**Review by DGamerL — CHANGES_REQUESTED** [review]
Wrong button, my bad

**Comment by Sirryan2002** [comment]
@wanderso

**Comment by DGamerL** [comment]
PR has been a few weeks since last activity, this PR will be closed for now.
Feel free to DM/Ping me on discord when you've made the changes and deconflicted the PR!

**Pull request closed without merging**